File: tests/lang/go/selfcheck/deliberate_failure_test.go
//go:build deliberate

// Run ONLY by `run-lang-tests.sh go --self-check` (which passes -tags deliberate).
// The build tag is what keeps a plain `go test ./...` from ever seeing this file, so the fixture
// can pass normally while still proving the lane can fail.
package selfcheck

import "testing"

func TestDeliberateFailure(t *testing.T) {
	t.Fatal("deliberate: this failure is the point")
}

File: tests/lang/java/src/test/java/lab/weyland/fixture/selfcheck/DeliberateFailureTest.java
package lab.weyland.fixture.selfcheck;

import org.junit.jupiter.api.Test;

/**
 * Run ONLY by {@code run-lang-tests.sh java --self-check} (Maven profile {@code selfcheck}).
 *
 * <p>MUST fail. A lane never seen failing is not a lane — the same argument the B148 guard makes
 * about itself. Surefire excludes this package by default, so a normal {@code mvn test} cannot
 * collect it.
 */
class DeliberateFailureTest {
    @Test
    void deliberateFailure() {
        throw new AssertionError("deliberate: this failure is the point");
    }
}
